Updates to fix buy button showing for nft owner, and fix CIP25 metadata for blockfrost

rynoV opened the pull request:
Remove connectWallet, replace with getWalletPkh, update getWalletBalance to return lovelace.

These changes are for https://github.com/mlabs-haskell/nft-marketplace/pull/230

Also includes changes described here: https://github.com/mlabs-haskell/seabug/pull/41

Also includes a fix to handle Blockfrost's recent change in handling CIP25 metadata, see:

- For context: https://github.com/Plutonomicon/cardano-transaction-lib/issues/1309
- The fix in CTL: https://github.com/Plutonomicon/cardano-transaction-lib/pull/1312

rynoV committed:
Updates to fix buy button showing for nft owner

Remove connectWallet, replace with getWalletPkh, update
getWalletBalance to return lovelace

rynoV committed:
Switch test log level to error

rynoV committed:
Update test report

rynoV committed:
Update test report commit names

aciceri committed:
Correct bundle for nft-marketplace

rynoV committed:
Don't require a wallet connection for all contracts

rynoV committed:
Fix getWalletBalance type

samuelWilliams99 commented on src/Seabug/CallContract.purs:
```diff
@@ -73,26 +74,27 @@ import Serialization.Hash
 callGetWalletBalance
-  :: ContractConfiguration -> Effect (Promise (Nullable Value))
+  :: ContractConfiguration -> Effect (Promise (Nullable BigInt))
 callGetWalletBalance cfg = Promise.fromAff do
```
This might be better as getWalletAda or getWalletLovelace
getBalance in other cardano contexts often means the full Value

rynoV committed:
Rename getWalletBalance to getWalletLovelace

rynoV committed:
Update test report

rynoV committed:
Update CIP25 metadata handling for recent Blockfrost changes

- Updates to a CTL revision which encodes the CIP25 metadata keys as
strings instead of bytes, for compatibility with Blockfrost
- Updates the Blockfrost response parsing for the new
`onchain_metadata` format

rynoV committed:
Update test report

samuelWilliams99 commented:
I believe so, but lets prehaps wait on Calum here

rynoV commented:
@aciceri  let's hold off for now, I may need to update the CTL revision again as changes were requested on my CTL PR

rynoV committed:
Update CTL version

rynoV committed:
Update test report

rynoV committed:
Merge branch 'master' of https://github.com/mlabs-haskell/seabug-contracts into calum/fixing-buy-button

rynoV commented:
Hey @aciceri, we should be good to merge whenever. I also pushed another commit to `calum/updates` on `seabug` to update the submodule versions.
